ci/docker/conda-cpp.dockerfile

```diff
@@ -73,6 +73,7 @@ ENV ARROW_BUILD_TESTS=ON \
     ARROW_PARQUET=ON \
     ARROW_PLASMA=ON \
     ARROW_S3=ON \
+    ARROW_GCS=ON \
```

Comment:
Can we keep alphebetical order? Same for other changes below.

Reply:
Thanks for reviewing! I've reordered them.